virtualenv.py

```diff
@@ -1185,26 +1185,55 @@ def change_prefix(filename, dst_prefix):
     assert False, "Filename {} does not start with any of these prefixes: {}".format(filename, prefixes)
 
 
-def copy_required_modules(dst_prefix, symlink):
-    import warnings
+if sys.version_info < (3, 4):
```

Comment:
I'm not a big fan of this kind of global switch, think to put the if within the function makes it easier to read.

Reply:
Might be so with servers and other high-performance code that calls the function multiple times, but here we're talking a process that 99% of the time loads module, executes and finishes. I think through such global ifs make the code hard to read and make you redefine the function header multiple time, which is error-prone and extra maintenance needed. We're talking either way about an import (a dictionary lookup, once already loaded), and an if check. Worrying about this here feels very much pre-mature optimization.

Reply:
I disagree, I think it's actually _less_ maintenance.  Right now when dropping an old version you simply delete the block of code -- if you intersperse it inside the function you have to perform surgery.  Also I think it makes more sense to always follow a practice instead of cherry picking on personal preference

Reply:
You haven't swayed me personally on this yet. Do you have any material that recommends this best practice with reasons why? Any other maintainers for virtualenv feel strongly enough to churn in on this? Otherwise I'm tempted to go with what feels easiest to me. I don't believe it's any easier, no surgery needed, you just remove an inner if branch rather than an outside one.